store.go

```diff
@@ -413,15 +414,32 @@ func (ls *LimitStore) Collect(id SpanID, anns ...Annotation) error {
 	ls.mu.Lock()
 	if ls.ring == nil {
 		ls.ring = make([]int64, ls.Max)
+		ls.traces = make(map[ID]struct{}, ls.Max)
 	}
+
+	// Check if the trace already exists in the ring. Otherwise, we would evict
+	// an old trace upon each annotation collection, rather than upon each new
+	// trace.
+	ids := make([]ID, 0)
+	for _, x := range ls.ring {
+		ids = append(ids, ID(x))
+	}
+	if _, ok := ls.traces[id.Trace]; ok {
+		ls.mu.Unlock()
+		return ls.DeleteStore.Collect(id, anns...)
+	}
+
 	if nextInsert := ls.ring[ls.nextInsertIdx]; nextInsert != 0 {
```

Comment:
Replace with:
```
_, exists := ls.traces[id.Trace]
if nextInsert := ls.ring[ls.nextInsertIdx]; nextInsert != 0 && !exists {
...
}
```

And then delete the above `if` clause that checks for `ls.traces[id.Trace]` and returns early. The code is clearer and more concise without the early return.